operator/anarchyrunner.py

```diff
@@ -103,64 +134,105 @@ def make_pod_template(self, runner_token=None):
         self.pods.pop(pod.metadata.name, None)
-        await self.manage_pods()
+        await self.manage_pods(logger=logger)
 
-    async def handle_runner_pod_deleting(self, pod):
+    async def handle_runner_pod_deleting(self, pod, logger):
         self.pods.pop(pod.metadata.name, None)
-        await self.manage_pods()
+        await self.manage_pods(logger=logger)
 
-    async def handle_runner_pod_labeled_for_temination(self, pod):
+    async def handle_runner_pod_labeled_for_temination(self, pod, logger):
         self.pods.pop(pod.metadata.name, None)
-        await self.manage_pods()
+        await self.manage_pods(logger=logger)
 
-    async def handle_runner_pod_event(self, pod):
+    async def handle_runner_pod_event(self, pod, logger):
         if pod.metadata.deletion_timestamp:
-            await self.handle_runner_pod_deleting(pod)
+            await self.handle_runner_pod_deleting(pod=pod, logger=logger)
         elif Anarchy.runner_terminating_label in pod.metadata.labels:
-            await self.handle_runner_pod_labeled_for_temination(pod)
+            await self.handle_runner_pod_labeled_for_temination(pod=pod, logger=logger)
         else:
-            await self.handle_runner_pod_update(pod)
+            await self.handle_runner_pod_update(pod=pod, logger=logger)
 
-    async def handle_runner_pod_labeled_for_termination(self, pod):
+    async def handle_runner_pod_labeled_for_termination(self, pod, logger):
         pass
 
-    async def handle_runner_pod_update(self, pod):
-        await self.manage_pod(pod)
+    async def handle_runner_pod_update(self, pod, logger):
+        await self.manage_pod(pod=pod, logger=logger)
 
-    async def handle_update(self):
+    async def handle_update(self, logger):
         await self.manage_service_account()
-        await self.manage_pods()
+        await self.manage_pods(logger=logger)
 
-    async def manage_pods(self):
+    async def manage_pods(self, logger):
         if not self.pods_preloaded:
             await self.preload_pods()
 
+        have_pending_pod = False
         for name, pod in list(self.pods.items()):
-            await self.manage_pod(pod)
+            await self.manage_pod(pod=pod, logger=logger)
+            if pod.status.phase == 'Pending':
+                have_pending_pod = True
 
         while len(self.pods) < self.min_replicas:
-            await self.create_runner_pod()
+            await self.create_runner_pod(logger=logger)
+
+        idle_runner_count = 0
+        for pod in self.status.get('pods', []):
+            if 'run' not in pod:
+                idle_runner_count += 1
+
+        if not have_pending_pod \
+        and self.scale_up_threshold \
+        and self.scale_up_threshold < len(self.status.get('pendingRuns', [])) - idle_runner_count \
+        and (not self.max_replicas or len(self.pods) < self.max_replicas) \
+        and (datetime.now(timezone.utc) - self.last_scale_up_datetime).total_seconds() > self.scale_up_delay:
```

Comment:
Should we simplify this condition by having plumbing methods?

```python
if not have_pending_pod \
        and self.scale_up_threshold \
        and self.reached_threshold() \
        and not self.reached_max_replicas() \
        and self.scale_up_delay_passed():
```

And probably add unit tests for those methods to validate logic and edge cases.